File: 230. Kth Smallest Element in a BST/kthSmallest.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def __init__(self):
        self.res = 0
        self.n = 0
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        self.n = k
        self.traverse(root)
        return self.res
    def traverse(self, root):
        if root == None or self.n < 0:
            return
        self.traverse(root.left)
        self.n = self.n - 1
        if self.n == 0:
            self.res = root.val
        self.traverse(root.right)

File: 230. Kth Smallest Element in a BST/kthSmallest.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int res = 0;
    int n = 0;
    public int kthSmallest(TreeNode root, int k) {
        n = k;
        traverse(root);
        return res;
    }
    public void traverse(TreeNode root) {
        if (root == null || n < 0) {
            return;
        }
        traverse(root.left);
        n = n - 1;
        if (n == 0) {
            res = root.val;
        }
        traverse(root.right);
    }
}